cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.wipro.ats.bdre.clustermigration;
+
+import com.wipro.ats.bdre.BaseStructure;
+import com.wipro.ats.bdre.IMConfig;
+import com.wipro.ats.bdre.clustermigration.beans.MigrationPreprocessorInfo;
+import com.wipro.ats.bdre.im.IMConstant;
+import com.wipro.ats.bdre.md.api.GetProperties;
+import com.wipro.ats.bdre.md.api.ProcessLog;
+import com.wipro.ats.bdre.md.beans.ProcessLogInfo;
+import org.apache.commons.cli.CommandLine;
+import org.apache.log4j.Logger;
+
+import java.sql.*;
+import java.util.*;
+import java.util.Date;
+
+/**
+ * Created by cloudera on 3/29/16.
+ */
+public class MigrationPreprocessor extends BaseStructure{
+    private static final Logger LOGGER = Logger.getLogger(MigrationPreprocessor.class);
+    private static String sourceRegularColumns=new String();   //comma-separated with datatype
+    private static String sourcePartitionColumns=new String(); //comma-separated with datatype
+    private static String stgPartitionColumns=new String();    //replacing source tech_partition with bdre tech partition
+    private static String stgTableDDL=new String();            //contains source stage table ddl
+    private static String destTableDDL=new String();           //contains destination table ddl
+    private static String filterCondition=new String();
+    private static String srcStgTableLocation=new String();
+    private static String destTableLocation=new String();
+    private static final String[][] PARAMS_STRUCTURE = {
+            {"pp", "parent-process-id", "Parent Process id of migration preprocessor"},
+            {"p", "process-id", " Process id of migration preprocessor"},
+            {"ied", "instance-exec-id", " instance exec id of preprocessor"}
+    };
+
+
+    protected static Connection getHiveJDBCConnection(String dbName, String hiveConnection) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00112)